### PR TITLE
Deploy to S3 Bucket

### PR DESCRIPTION
**Pull request by BMO-tech**
**Summary of change:**
Updates to the `.circleci/config.yml` file to introduce a build and deploy process. This process will create a `.zip` artifact of the theme, a `<theme-name>.<version>.json` file, and upload these items to the appropriate S3 bucket for the Product Info Service to consume.

**See:** GF-1739 Internal WPE Ticket

### CODE REVIEW

**Comment by BMO-tech** [comment]
Notes from the previous PR:
* `.circleci/config.yml:93-94`: The jobs that this job is dependent on might not run on a tag by default. You will probably have to add the tags: filter to 'checkout' and 'standards' also.
  * Fixed
* `.circleci/config.yml:96-99`: Any reason to build the zip bundle only on master and not on every PR? Having that zip there to download is a way someone on the team could download what was built and test it.
  * This is to prevent any PR from triggering a full Build job. This is a public repo where anyone can contribute by creating a PR.
* `.circleci/config.yml:82`: Use Atomic Blocks Pro CI/CD S3 Bucket
  * Fixed

File: .circleci/config.yml [hunk 1]
```diff
@@ -1,42 +1,128 @@
 version: 2.1
 
-jobs:
-  standards:
+commands:
+  install-dependencies:
```
Comment (markkelnar) [inline .circleci/config.yml]: I so see the dashes vs underscores difference between this command and the one you added.  I know this command already existed, you just moved things around.  Part of me says let's pick one way.  I prefer underscores for these commands.  You're call if you change it in this PR.

Comment (BMO-tech) [reply .circleci/config.yml]: Good call, consistency is important. I'll see if there is a CircleCI preferred standard, if not then I'll standardize one way or the other with reasons for why.

Comment (BMO-tech) [reply .circleci/config.yml]: Alright, decided to go with `snake_case` as the rest of Circle's default commands follow this pattern. (IE: `attach_workspace` and `persist_to_workspace`)

**Comment by nickcernis** [comment]
@BMO-tech Latest update looks good.

**Pull request merged**